src/navigation/components/MenuDetailsPage/MenuDetailsPage.tsx

```diff
@@ -69,7 +69,7 @@ const MenuDetailsPage: React.FC<MenuDetailsPageProps> = ({
     });
 
     if (result) {
-      setTreeOperations([]);
+      setTreeOperations([...treeOperations]);
```

Comment:
I don't see how that's doing anything, you're setting current state as new state.

Reply:
With empty array, as it was before, navigation items would move back to their previous location. 
I realized that the if statement block there is redundant - I will remove it next time.